src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -131,11 +131,25 @@ protected override void CheckInterfaces(DiagnosticBag diagnostics)
                 var corLibrary = this.ContainingAssembly.CorLibrary;
                 var conversions = new TypeConversions(corLibrary);
                 var location = singleDeclaration.NameLocation;
+                var seenInterfaces = PooledHashSet<NamedTypeSymbol>.GetInstance();
 
                 foreach (var @interface in interfaces)
                 {
                     @interface.CheckAllConstraints(conversions, location, diagnostics);
+
+                    foreach (var seenInterface in seenInterfaces)
+                    {
+                        if (@interface.Equals(seenInterface, TypeCompareKind.IgnoreTupleNames))
```

Comment:
I debated on that. In most cases it should be a smaller number, in which case I preferred to avoid the allocation. I don't have an perf data to support though.
What's a good way to decide this one? Any micro-benchmark I try probably won't be representative...